### ISSUE TITLE
The mouse pointer for tables is tiny (not scaled) on HDPI screens.

### ISSUE THREAD

**Issue by PJSingh5000** [ONLYOFFICE/DesktopEditors#566]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
The mouse pointer for tables is tiny (not scaled) on HDPI screens.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
1. Launch ONLYOFFICE desktop editors on a HDPI screen (ex 3848x2160 with 200% scaling).
2. Create a new document.
3. Create a table.
4. Move the cursor above a column to select the column.
5. _Notice that the mouse pointer becomes very tiny (not scaled)._
6. Move the cursor near a cell to select the cell.
7. _Notice that the mouse pointer becomes very tiny (not scaled)._
8. Move the cursor to the left of a row to select the row.
9. _Notice that the mouse pointer becomes very tiny (not scaled)._

Images of tiny mouse pointers (from steps 5, 7, and 9 above), circled in red.
![image](https://user-images.githubusercontent.com/19394936/107892387-ba335780-6ef2-11eb-9507-5e193a596a01.png)

For comparison, notice that the insert cursor is correctly scaled, circled in red.
![image](https://user-images.githubusercontent.com/19394936/107892554-e69ba380-6ef3-11eb-820a-b7234febdab1.png)

**What is the expected behavior?**
The mouse pointer scaling when handling tables should match the scaling set by the OS, as is done for the insert cursor.

**Which versions of DesktopEditors, and which OS are affected by this issue? Did this work in previous versions of DesktopEditors?**

OS: Ubuntu 20.04.2 LTS
Software: onlyoffice-desktopeditors version 6.1.0-90

This did not work in previous versions.

**Comment by ShockwaveNN** [comment]
I've got a 3848x2160
And for me cursor size is the same for default cursor and for table cursor
![image](https://user-images.githubusercontent.com/668524/107921028-eeecf080-6f7e-11eb-934f-a2add18826c8.png)
![image](https://user-images.githubusercontent.com/668524/107921055-fd3b0c80-6f7e-11eb-982c-57baddadb98d.png)

Maybe the reason that you have 200% scaling, I think I got 100% scaling on my system, but not sure how to change it

Could you try to set 100% scaling on your system and check that too?

**Comment by PJSingh5000** [comment]
I am using Ubuntu 20.04 (GNOME Desktop) on a 3848x2160 screen.

The OS automatically detects the screen size and offers scale option under the Display section of the Settings app.

![image](https://user-images.githubusercontent.com/19394936/108216149-23d48100-7100-11eb-8b9d-7cb4a4927da5.png)

I found that the UI for ONLYOFFICE doesn't scale at all when I change the desktop scale.

**(1) Screenshot using 100% scaling.**

This is a table created in ONLY office using 100% scaling.

Note the size of the "+" icon, the insert bar "|", and the down arrow.

![image](https://user-images.githubusercontent.com/19394936/108216671-bb39d400-7100-11eb-844d-62b40b2a2e67.png)

Notice the relative sizes of the insert bar "|" in ONLYOFFICE the insert cursor "I" rendered by the OS. (The OS cursor is small because the OS is using 100% scaling). Also, note the small height of the title bar and the small size of the font in the title bar, because the OS is set to 100% scaling.

![image](https://user-images.githubusercontent.com/19394936/108219357-aa3e9200-7103-11eb-8888-976e554631f2.png)


**(2) Screenshot using 200% scaling.**

This is a table created in ONLY office using 200% scaling.

Note the size of the "+" icon, the insert bar "|", and the down arrow.

The sizes of these items @200% scale are the same as in screenshot above @100% scale.

(I did restart ONLY office between tests).

![image](https://user-images.githubusercontent.com/19394936/108216893-f6d49e00-7100-11eb-9fbd-03a35fe756c0.png)

Notice the relative sizes of the insert bar "|" in ONLYOFFICE the insert cursor "I" rendered by the OS. (The OS cursor is larger because the OS is using 200% scaling). Also, note the large height of the title bar and the large size of the font in the title bar, because the OS is set to 200% scaling.

![image](https://user-images.githubusercontent.com/19394936/108219505-cb9f7e00-7103-11eb-8b44-a897f4a3272a.png)

**Questions**

1. Does ONLYOFFICE have only one scale, regardless of what is selected in the OS?

2. Otherwise, is does ONLYOFFICE scale itself, independent of the setting selected in the OS.

Nevertheless, the arrows used with tables in ONLYOFFICE are very tiny and hard to see on HDPI screens, but they would be OK for non-HDPI screens.

**Comment by ShockwaveNN** [comment]
> Does ONLYOFFICE have only one scale, regardless of what is selected in the OS?

No, we got several scales, you try to play with values mentioned in [this comment](https://github.com/ONLYOFFICE/DesktopEditors/issues/217#issuecomment-651705164)

> Otherwise, does ONLYOFFICE scale itself (by detecting screen resolution), independent of the setting selected in the OS?

I think our code try to calculate scaling setting on ourself, not using system values. We're aware this aloritm is not 100% excellent, and got several issues including https://github.com/ONLYOFFICE/DesktopEditors/issues/217

**Comment by PJSingh5000** [comment]
Thanks for your quick response.

I'll try the options you linked to.

Your answer to my second question explains a lot hand makes sense.

Actually, the algorithm is quite good. (I remember years ago when ONLYOFFICE didn't scale at all, so it is quite usable with the current algorithm).

For this bug report, one improvement that ONLYOFFICE could make, if possible, is to apply the scaling algorithm to the "table selection arrows". This is already done for the insert bar "|".

**Comment by ShockwaveNN** [comment]
@PJSingh5000 Thank you for your report, I've found a way how it can be easily reproduced

In my kubuntu I've got this setting for cursor size:
![image](https://user-images.githubusercontent.com/668524/108225405-f6fa8b00-714c-11eb-86cf-b2fbcd28ad0a.png)

And changing this setting to any value will not change size of cursor for "table selection arrows"
So I've created a bug 48779 in our private issue tracker to discuss this

**Comment by il2020** [comment]
I have a similar problem.


https://github.com/ONLYOFFICE/DesktopEditors/assets/77854832/24e5dca9-0eb1-4db0-b6ea-e63e441e3a58

**Comment by PJ-Singh-001** [comment]
I am still experiencing in Ubuntu  23.10.

**Comment by Soljanka** [comment]
Can confirm: it's still an issue on DesktopEditors 8.2.2.22 (deb)
Easily reproducable on Linux Mint 22.1 with fractional scaling

steps to reproduce:
1) enable fractional scaling

![Image](https://github.com/user-attachments/assets/d7388d01-7bab-4b78-a97f-572c634e1020)

2) open any spreadsheet

3) hovering over the menu items - mouse pointer still looks fine

![Image](https://github.com/user-attachments/assets/f263d799-070c-4fd6-a15d-95ce43ded6fd)

4) but not anymore as soon as the mouse pointer is inside a cell

![Image](https://github.com/user-attachments/assets/4274a8ba-757f-4fed-9ebb-621ed779bd91)

**Comment by PJ-Singh-001** [comment]
This is still an issue in ONLYOFFICE 9.0.

**Comment by gleb-akhmerov** [comment]
Still an issue in 9.3.1.8. Was there ever a fix for this? Or has this been a problem since February 2021, for the past 5 years?

**Comment by stochmalm** [comment]
Still the same problem, using latest cachyos + GNOME 49 on wayland. I'm not even using fractional scaling, but "big text" from accessibility settings on 4k display.

**Comment by regs01** [comment]
9.3.1 the problem is there in Windows as well

non-system cursors are barely visible

**Comment by TomAn-Ydx** [comment]
Stil the same issue in Sep 2026.
Any pointers where we may edit in the code to fix this? Don't mind spending some time on it.
